src/libraries/System.Private.CoreLib/src/System/Globalization/EastAsianLunisolarCalendar.cs

```diff
@@ -646,15 +646,15 @@ public override bool IsLeapYear(int year, int era)
             return InternalIsLeapYear(year);
         }
 
-        private const int DefaultGregorianTwoDigitYearMax = 2029;
+        private const int DefaultGregorianTwoDigitYearMax = 2049;
 
         public override int TwoDigitYearMax
         {
             get
             {
                 if (_twoDigitYearMax == -1)
                 {
-                    _twoDigitYearMax = GetSystemTwoDigitYearSetting(BaseCalendarID, GetYear(new DateTime(DefaultGregorianTwoDigitYearMax, 1, 1)));
+                    _twoDigitYearMax = GetSystemTwoDigitYearSetting(ID, DefaultGregorianTwoDigitYearMax);
```

Comment:
Why did you change this? `GetYear` gets the year from the used calendar and not necessary to match the Gregorian year.

Reply:
that was my bad.